ctapipe/ml/stereo_combination.py

```diff
@@ -44,19 +47,102 @@ class StereoMeanCombiner(StereoCombiner):
     the reconstruction on telescope-level with an optional weighting.
     """
 
-    def __init__(self, mono_prediction_column, weight_column=None, *args, **kwargs):
-        super().__init__(mono_prediction_column=mono_prediction_column, *args, **kwargs)
-        self.weight_column = weight_column
+    weights = CaselessStrEnum(
+        ["none", "intensity", "konrad"], default_value="none"
+    ).tag(config=True)
+    combine_property = CaselessStrEnum(["energy", "classification", "direction"]).tag(
+        config=True
+    )
 
-    def _construct_weights(self, table):
-        # TODO: Do we want to calculate weights on the fly or do it externally?
-        # If we dont want to do that here, this might as well be removed
-        if self.weight_column:
-            return table[self.weight_column]
-        else:
-            return None
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-    def __call__(self, mono_predictions: Table) -> Table:
+    def _calculate_weights(self, dl1):
+        """ """
+        if isinstance(dl1, Container):
+            if self.weights == "intensity":
+                return dl1.hillas["intensity"]
+            elif self.weights == "konrad":
+                return (
+                    dl1.hillas["intensity"] * dl1.hillas["length"] / dl1.hillas["width"]
+                )
+            else:
+                return 1
+        elif isinstance(dl1, Table):
+            if self.weights == "intensity":
+                return dl1["hillas_intensity"]
+            elif self.weights == "konrad":
+                return (
+                    dl1["hillas_intensity"] * dl1["hillas_length"] / dl1["hillas_width"]
+                )
+            else:
+                return np.ones(len(dl1))
+        return 1
+
+    def _weighted_mean_ufunc(self, tel_values, weights, n_array_events, indices):
```

Comment:
Why is this a method and the other one a free function? This could at least be a `staticmethod` since it's not acutally using `self`.